utils/core/src/serde/mod.rs

```diff
@@ -197,3 +279,160 @@ pub trait Deserializable: Sized {
         Ok(result)
     }
 }
+
+impl Deserializable for () {
+    fn read_from<R: ByteReader>(_source: &mut R) -> Result<Self, DeserializationError> {
+        Ok(())
+    }
+}
+
+impl<T1> Deserializable for (T1,)
```

Comment:
minor nit: do we need the comma after `T1`?

Reply:
https://github.com/facebook/winterfell/pull/233#discussion_r1417277215